packages/powersync/lib/src/powersync_database.dart

```diff
@@ -98,18 +102,55 @@ class PowerSyncDatabase with SqliteQueries implements SqliteConnection {
   /// additional logic to run inside the database isolate before or after opening.
   ///
   /// Subclass [PowerSyncOpenFactory] to add custom logic to this process.
-  factory PowerSyncDatabase.withFactory(PowerSyncOpenFactory openFactory,
-      {required Schema schema,
-      int maxReaders = SqliteDatabase.defaultMaxReaders}) {
+  factory PowerSyncDatabase.withFactory(
+    PowerSyncOpenFactory openFactory, {
+    required Schema schema,
+    int maxReaders = SqliteDatabase.defaultMaxReaders,
+    LogType log = LogType.auto,
+  }) {
     final db = SqliteDatabase.withFactory(openFactory, maxReaders: maxReaders);
-    return PowerSyncDatabase.withDatabase(schema: schema, database: db);
+    return PowerSyncDatabase.withDatabase(
+        schema: schema, database: db, log: log);
   }
 
   /// Open a PowerSyncDatabase on an existing [SqliteDatabase].
   ///
   /// Migrations are run on the database when this constructor is called.
-  PowerSyncDatabase.withDatabase(
-      {required this.schema, required this.database}) {
+  PowerSyncDatabase.withDatabase({
+    required this.schema,
+    required this.database,
+    LogType log = LogType.auto,
+  }) {
+    if (log == LogType.debug || log == LogType.auto) {
```

Comment:
can we extract this into a method that returns the logger?

Reply:
I'm thinking it might be even better to pass in a Logger instance rather than the enum - will check if that's feasible